src/web/error.rs

```diff
@@ -113,11 +113,37 @@ mod tests {
     use kuchiki::traits::TendrilSink;
 
     #[test]
-    fn check_404_page_content() {
+    fn check_404_page_content_crate() {
         wrapper(|env| {
             let page = kuchiki::parse_html().one(
                 env.frontend()
-                    .get("/page-which-doesnt-exist")
+                    .get("/crate-which-doesnt-exist")
+                    .send()?
+                    .text()?,
+            );
+            assert_eq!(page.select("#crate-title").unwrap().count(), 1);
+            assert_eq!(
+                page.select("#crate-title")
+                    .unwrap()
+                    .next()
+                    .unwrap()
+                    .text_contents(),
+                "The requested crate does not exist",
+            );
+
+            Ok(())
+        });
+    }
+
+    #[test]
+    fn check_404_page_content_resource() {
+        // Resources with a `.js` and `.ico` extension are special cased in the
+        // routes_handler which is currently run last. This means that `get("resource.exe")` will
+        // fail with a `no so such crate` instead of 'no such resource'
```

Comment:
Ooh, good catch. This seems like an ok bug to have though - if you're trying to download .exes docs.rs doesn't want to help you even as much to have the right error.